### PR TITLE
Feature/update orders table

### LINKED ISSUE #8806: Orders list not reloaded correctly after cancelling an order
## Description
We had two issues around cancelling orders recently: #8620 and #8704. Both seem to be fixed now, even though there were no dedicated pull requests for them. But now, after cancelling an order, the list of orders remains blank and needs to be reloaded manually. By clicking on the 'orders' tab the OLD list is displayed, with the cancelled order still present. Only refreshing the whole page displays the updated list. 
Before, the page automatically displayed the remaining orders after a short moment.

## Expected Behavior
When cancelling an order in the user account, the remaining orders should be displayed automatically.

## Actual Behaviour
When cancelling an order in the user account, the page displays no orders at all and the old list of orders when clicking the 'orders tab'.

## Steps to Reproduce
1. As a shop owner open an order cycle and allow orders to be changed or canceled (/admin/enterprises/shop/edit#!/shop_preferences)
2. As a customer place at least two orders
3. Cancel one of the orders in the user account
4. Observe the empty page
5. Click the 'orders' tab
6. Observe the old list of orders
7. Refresh the page
8. Observe the updated list of orders

## Animated Gif/Screenshot

https://user-images.githubusercontent.com/1790176/151640167-e6219058-4d4c-4852-b8b9-6b9a0cefce84.mp4

## Workaround
Reload manually.

## Severity
bug-s4: it's annoying, but you can use it

## Your Environment
* Version used: v4.1.14
* Browser name and version: Firefox 96.0 (64-Bit)
* Operating System and version (desktop or mobile): Ubuntu 20.04 LTS Desktop

### LINKED ISSUE #9304: As a shopper, I can have a better feedback on cancelled orders
## Description
**- As a:** shopper
**- On page:** `/account#/orders`
**- I want to be able to do:** See when orders are cancelled

-  Replace the "view" column by "status" column,
-  Add a link on confirmed status only, to view the order.
-  Remove links from order numbers
-  Add a flash error after cancelling

Previously: 

New mockup:

also note the transaction tab is out of scope here.

## Acceptance Criteria & Tests

*Scenario 1 : the shopper just visit the page*

1. Cancelled orders are visible and cannot be displayed
2.  Completed orders can be displayed by clicking on the status
3. Links are removed from order number

*Scenario 2 : the shopper is the one cancelling the order*

1. As a shopper cancel an order
2. See a flash message confirming the order is completed
3. See the correct status on your account
4. check that you still get the usual cancelling email.

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?
This issue makes light changes in the account#/orders page and renders a flash when orders are cancelled.
Closes #9304 
Closes #8806 

#### What should we test?

- Visit page account/orders
- Cancel an Open Order
- Ascertain that a flash message is displayed
---
- Ascertain that the Order Number is no longer a link
- Ascertain that the view column has been renamed to status
- Ascertain that only orders with Completed status have a link to the orders#show page
- Ascertain that orders with Cancelled status don't have a link to the orders#show page

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

SS:

ezgif com-gif-maker(8)

### CODE REVIEW

**Comment by binarygit** [comment]
Hey @jibees looks like some tests are failing. I was not sure if I screwed something up...and still believe that I might have screwed something up.
But looking at some of the failures makes me think that it was because some of them are checking for `.status` column to be present in the past orders table:
```
Failures:
  1) Authentication viewing my account
Failure/Error: %th.order7.text-right= t('.status')
ActionView::Template::Error:
  translation missing: en.spree.users.past_orders.status
```

```
8) 
    As a consumer
    I want to view my order history with each hub
    and view any outstanding balance.
 as a logged in user without any completed orders displays an appropriate message
Failure/Error: %th.order7.text-right= t('.status')
ActionView::Template::Error:
  translation missing: en.spree.users.past_orders.status
  ```
  So I was wondering if I should go into the tests and try to fix them. Or if It's me that has just screwed things up.

**Comment by jibees** [comment]
As far as I read, there are some missing i18n keys. 
You've added in your PR: `t('.status')` but no key could be found under: `en.spree.users.past_orders.status`. You should add them into the `en.yml` file.

**Comment by jibees** [comment]
also, `spec/controllers/spree/orders_controller_spec.rb` fails (on line 458 and 475) probably because you've changed `app/controllers/spree/orders_controller.rb`. It's up to you to change the controller to match the spec nor to change the specs to match the changes you've made on the controller (I don't know, I didnt' read deep).

**Comment by binarygit** [comment]
Hey @jibees just wanted to give you an update. Most of the errors went away after adding the status key in en.yml (Thank you for that! :fireworks: :pray: ). There was another test that broke because it was checking if Open Orders table had a column with the Order Num as a link, I changed that one.

Now, there are only two tests which are breaking. They're from the orders controller and it's an unusual error message that I've been unable to find a solution for.
I describe it here: https://openfoodnetwork.slack.com/archives/C2GQ45KNU/p1657431656609309

Let me know if you've seen this before and can help me out. Cheers!

**Comment by binarygit** [comment]
Hello @jibees. I apologize for taking some time with this. The error in the orders_controller test was very unusual so it took me a little time to figure out what to do about it. Anyway, have a look and let me know if there's anything. :smile:

File: app/views/spree/users/_past_orders.html.haml [hunk 1]
```diff
@@ -8,16 +8,16 @@
...
             %a{"ng-href" => "{{::Orders.shopsByID[order.shop_id].hash}}#{main_app.shop_path}", "ng-bind" => "::Orders.shopsByID[order.shop_id].name"}
           %td.order3.show-for-large-up{"ng-bind" => "::order.completed_at"}
           %td.order4.show-for-large-up{"ng-bind" => "::order.item_count"}
           %td.order5.text-right{"ng-class" => "{'debit': order.payment_state != 'paid', 'credit': order.payment_state == 'paid'}","ng-bind" => "::order.total | localizeCurrency"}
           %td.order6.text-right.show-for-large-up{"ng-class" => "{'debit': order.payment_state != 'paid', 'credit': order.payment_state == 'paid'}", "ng-bind" => "::(order.payment_state == 'paid' ? 'say_yes' : 'say_no') | t"}
-          %td.order7.text-right
-            %a{"ng-href" => "{{::order.path}}" }= t('.view')
+            %td.order7.text-right{ "ng-switch" => "::order.state"  }
+              %a{ "ng-switch-when" => "complete", "ng-href" => "{{::order.path}}" }Completed
+              %span{ "ng-switch-when" => "canceled" }Cancelled
```
Comment (mkllnk) [inline app/views/spree/users/_past_orders.html.haml:23]: Completed and Cancelled should also be translatable.
```suggestion
              %a{ "ng-switch-when" => "complete", "ng-href" => "{{::order.path}}" }= t(".completed")
              %span{ "ng-switch-when" => "canceled" }= t(".cancelled")
```
Plus the corresponding additions to en.yml.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Thank you for these improvements.

File: app/views/spree/users/_past_orders.html.haml (hunk 1 above)
Comment (binarygit) [reply app/views/spree/users/_past_orders.html.haml]: Hey I made the requested changes but forgot to reply here! Could you check and merge this if everything's inorder or let me know if something else is required.

**Review by mkllnk — APPROVED** [review]
Thanks!

**Comment by drummer83** [comment]
## Expected Behavior
### From #8806
- When cancelling an order in the user account, the remaining orders should be displayed automatically. --> OK (see screen capture below)

### From #9304
- Replace the "view" column by "status" column --> OK
- Add a link on confirmed status only, to view the order --> OK
    - Orders R868323861 and R072768328 show the snail --> not introduced by this PR
- Remove links from order numbers --> OK
- Add a flash error after cancelling --> OK
    - Orders R415513613, R840415781, R038885725 cannot be cancelled --> not introduced by this PR, but after the PR there is at least a flash error telling that the order couldn't be cancelled (see screen capture)
    Peek 2022-08-01 20-05


**Scenario 1:** the shopper just visits the page
- Cancelled orders are visible and cannot be displayed --> OK
- Completed orders can be displayed by clicking on the status --> OK
- Links are removed from order number --> OK
Peek 2022-08-01 20-19

**Scenario 2:** the shopper is the one cancelling the order
- As a shopper cancel an order --> OK
- See a flash message confirming the order is cancelled --> OK
- See the correct status on your account --> OK
- Check that you still get the usual cancelling email --> OK
Peek 2022-08-01 20-08

## Summary

The strange behavior we see around some orders was not introduced by this PR. Everything else seems to work smoothly. Updating the orders list automatically is just awesome! :partying_face: 

This one is 'Ready to go'! :rocket: 
Thanks again, @binarygit!

**Pull request merged**